pokemon/damage.js

```diff
@@ -0,0 +1,168 @@
+    this._pName = pName;
+    this._pType = pType;
+    this._pAttack = pAttack;
+    this._pDefense = pDefense;
+  }
+
+  get pName() {
+    return this._pName;
+  }
+  get pType() {
+    return this._pType;
+  }   
+
+  get pAttack(){
+    return this._pAttack;
+  }
+
+  get pDefense(){
+    return this._pDefense
+  }
+}
+
+
+function damage(pPokemon1,pPokemon2){
+  let resultdamage = 50 * (pPokemon1.pAttack/pPokemon2.pDefense) * effective(pPokemon1.pType,pPokemon2.pType);
+  return resultdamage;
+}
+
+
+
+function getEffectivenessPoints(level){
+  var point;
+   function isHighEffectiveness () {
+     return point = 2;
+   }
+   function isNeutralEffectiveness () {
+     return point = 1;
+   }
+   function isLittleEffectiveness () {
+     return point = 0.5;
+   }
+
+   var points = {
+     'High' : isHighEffectiveness,
+     'Neutral' : isNeutralEffectiveness,
+     'Little' : isLittleEffectiveness,
+   };
+
+   return points[level]();
+}
+
+
+
+function effective(pTypePokemon1,pTypePokemon2){
+  var pokemonEffectiveness
+
+  var pokemonType = {
+    'fire-grass': function () {
+      pokemonEffectiveness = getEffectivenessPoints('High');
+    },
+    'fire-water': function () {
+      pokemonEffectiveness = getEffectivenessPoints('Little');
+    },
+    'fire-fire': function () {
+      pokemonEffectiveness = getEffectivenessPoints('Little');
+    },
+    'fire-electric': function () {
+      pokemonEffectiveness = getEffectivenessPoints('Neutral');
+    },
+    'grass-fire': function () {
+      pokemonEffectiveness = getEffectivenessPoints('Little');
+    },
+    'grass-grass': function () {
+      pokemonEffectiveness = getEffectivenessPoints('High');
+    },
+    'grass-water': function () {
+      pokemonEffectiveness = getEffectivenessPoints('High');
+    },
+    'grass-electric': function () {
+      pokemonEffectiveness = getEffectivenessPoints('Neutral');
+    },
+    'water-fire': function () {
+      pokemonEffectiveness = getEffectivenessPoints('High');
+    },
+    'water-grass': function () {
+      pokemonEffectiveness = getEffectivenessPoints('Little');
+    },
+    'water-water': function () {
+      pokemonEffectiveness = getEffectivenessPoints('Little');
+    },
+    'water-electric': function () {
+      pokemonEffectiveness = getEffectivenessPoints('Neutral');
+    },
+    'electric-fire': function () {
+      pokemonEffectiveness = getEffectivenessPoints('Neutral');
+    },
+    'electric-water': function () {
+      pokemonEffectiveness = getEffectivenessPoints('High');
+    },
+    'electric-grass': function () {
+      pokemonEffectiveness = getEffectivenessPoints('Little');
+    },
+    'electric-electric': function () {
+      pokemonEffectiveness = getEffectivenessPoints('Little');
+    },
+    'default' : function () {
+      pokemonEffectiveness = 'Sorry, the type of Pokemon dont exist';
+    }
+  };
+  (pokemonType[pTypePokemon1+"-"+pTypePokemon2] || pokemonType['default'])();
```

Comment:
Prefer to use spaces between operators in `pTypePokemon1 + "-" + pTypePokemon2`. However, use string template literals instead (https://developers.google.com/web/updates/2015/01/ES6-Template-Strings):
```
(pokemonType[`${pTypePokemon1}-${pTypePokemon2}`] || pokemonType['default'])();
```